perses/annihilation/relative.py

```diff
@@ -2541,3 +2537,269 @@ def _handle_exceptions(self):
+        self._transcribe_angles()
+        self._transcribe_torsions()
+
+        if 'NonbondedForce' in self._old_system_forces or 'NonbondedForce' in self._new_system_forces:
+            self._transcribe_nonbonded()
+            self._transcribe_exceptions()
+
+        # Get positions for the hybrid
+        self._hybrid_positions = self._compute_hybrid_positions()
+
+        # Generate the topology representation
+        self._hybrid_topology = self._create_topology()
+
+        # define the scale and nonscale region
+        self._make_scale_regions(scale_region)
+
+    def get_scale_identifier(self,
+                             particles):
+        """
+        return a bool based on whether the particles in a set are inside/outside/or straddling an alchemical region;
+        if nonbonded, particles is a hybrid integer, 1 indicates it is in scale region. otherwise, 0;
+
+        if bonded, particles is a hybrid-indexed set, [1,0,0] indicates scale region, [0,1,0] otherwise. if it straddles the region, gives [0,0,1].
+        """
+        assert type(particles) in [type(set()), int], f"`particles` must be an integer or a set, got {type(particles)}."
+        if type(particles) == int:
+            out = 1 if particles in self._scale_region else 0
+            return out
+
+        if particles.issubset(self._scale_region):
+            out = [1,0,0]
+        elif particles.issubset(self._nonscale_region):
+            out = [0,1,0]
+        else:
+            assert particles.issubset(self._scale_region.union(self._nonscale_region)), f"the union of the scale region and nonregion is all regions"
+            out = [0,0,1]
+        return out
+
+    def get_alch_identifier(self,
+                            particles):
+        """
+        return a list of integers that specifies whether a term is environment or alchemical (if alchemical, specifies which region, too)
+        """
+        template = [0 for i in range(self._topology_proposal._num_alchemical_regions + 1)]
+        template[0] = 1
+        for idx, _set in enumerate(self._topology_proposal._alchemical_regions):
+            if particles.intersection(_set) != {}: #allow for particles that straddle alchemical regions
+                template[idx+1] += 1 #update the template
+                template[0] = 0 #remove env default
+
+        assert sum(template) == 1, f"the alchemical regions must be disjoint; got {template}"
+        return template
+
+    @staticmethod
+    def render_bool_string(bool_variable_prefixes, bool_variable_names):
+        selections = [f"{bool_variable_prefix} * {bool_variable_name}" for bool_variable_prefix, bool_variable_name in zip(bool_variable_prefixes, bool_variable_names)]
+        return '( ' + '+'.join(selections) + ' )'
+
+
+    def _transcribe_bonds(self):
+        """
+        handle the harmonic bonds...
+        """
+        scale_bool_string = RxnHybridTopologyFactory.render_bool_string(['1', 'scale_lambda', 'interscale_lambda'],
```

Comment:
We should allow for multiple, disjoint scale regions. 
The disjoint-ness should be checked in the init()